File: sdkwork-app-sdk-typescript/src/api/list.ts
import { appApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { PlusApiResultPageOrganizationVO } from '../types';


export class ListApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** 获取组织列表 */
  async getOrganization(params?: QueryParams): Promise<PlusApiResultPageOrganizationVO> {
    return this.client.get<PlusApiResultPageOrganizationVO>(appApiPath(`/organization/list`), params);
  }
}

export function createListApi(client: HttpClient): ListApi {
  return new ListApi(client);
}

File: sdkwork-app-sdk-go/api/list.go
package api

import (
    sdktypes "github.com/sdkwork/app-sdk/types"
    sdkhttp "github.com/sdkwork/app-sdk/http"
)

type ListApi struct {
    client *sdkhttp.Client
}

func NewListApi(client *sdkhttp.Client) *ListApi {
    return &ListApi{client: client}
}

// 获取组织列表
func (a *ListApi) GetOrganization(query map[string]interface{}) (sdktypes.PlusApiResultPageOrganizationVO, error) {
    raw, err := a.client.Get(AppApiPath("/organization/list"), query, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultPageOrganizationVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPageOrganizationVO](raw)
}
